### PR TITLE
Enhancement/4092 analytics gathering data

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue #4086

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/datastore/report.js [hunk 1]
```diff
@@ -274,6 +277,42 @@ const baseSelectors = {
...
+			args.url = url;
+		}
+
+		const report = select( MODULES_ANALYTICS ).getReport( args );
+
+		if ( report === undefined ) {
+			return undefined;
+		}
+
+		if ( Array.isArray( report ) && report.length === 0 ) {
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/datastore/report.js:310]: I think the condition here should be if **not** an array **or** an empty array then return `true`.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @kuasha420. This looks solid to me. Have one comment and then it should be good to go.

File: assets/js/modules/analytics/datastore/report.js (hunk 1 above)
Comment (kuasha420) [reply assets/js/modules/analytics/datastore/report.js:310]: @eugene-manuilov really? Is my reading comprehension this bad or the IB is wrong? Can you have a look at the IB of this ticket?

Comment (eugene-manuilov) [reply assets/js/modules/analytics/datastore/report.js:310]: @kuasha420 Yes, IB says that it should only check whether it is an empty array, but we should also return `true` if the report data is `NULL` which is possible if I remember correctly.

Could you please check failed e2e and vrt checks?

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Pull request merged**